### PR TITLE
Wallet Balance Alert: Health

### LINKED ISSUE #435: Add Wallet Balance Alert to Health Endpoint
- [ ] Add wallet balance alert threshold to the detailed health endpoint (@jwicks) when the balance drops below a pre-defined (env var) bitcoin amount.

### PR DESCRIPTION
**Pull request by jwicks31**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Resolves #435 

- Added `addBalanceLowBool` to `refreshWalletInfo` in `src/Health/HealthController`
- Added `isBalanceLow` function with unit tests

### CODE REVIEW

File: src/Health/HealthController.ts [hunk 1]
```diff
@@ -98,6 +107,13 @@ export class HealthController {
     return pickWalletInfoKeys(await this.bitcoinCore.getWalletInfo())
   }
 
+  private async checkWalletBalance(walletInfo: WalletInfo): Promise<WalletInfo> {
+    const { balance } = walletInfo
+    return balance < this.configuration.lowWalletBalanceBTC
```
Comment (kennylavender) [inline src/Health/HealthController.ts]: Are we exposing the balance of the wallet? We probably shouldn't. The bool of balanceLow is probably all we should expose.

File: src/Health/HealthController.ts [hunk 2]
```diff
@@ -98,6 +107,13 @@ export class HealthController {
     return pickWalletInfoKeys(await this.bitcoinCore.getWalletInfo())
   }
 
+  private async checkWalletBalance(walletInfo: WalletInfo): Promise<WalletInfo> {
```
Comment (kennylavender) [inline src/Health/HealthController.ts]: This looks like a pure function that could be unit tested. Name could be better too, try to think through what this functions true responsibility is.

File: src/Health/HealthController.ts (hunk 1 above)
Comment (lautarodragan) [reply src/Health/HealthController.ts]: No no, it's ok to expose the balance of the wallet. The node won't be public in any environment for us and shouldn't for anyone else. 

None should set up a node directly against the public internet.

File: src/Health/HealthController.ts (hunk 2 above)
Comment (jwicks31) [reply src/Health/HealthController.ts]: Ok I updated the name and moved the checking logic to another function with unit tests. Looking better?

File: src/Health/HealthController.ts (hunk 1 above)
Comment (kennylavender) [reply src/Health/HealthController.ts]: Ok sounds good, thanks!

**Comment by jwicks31** [comment]
@kennylavender should I go ahead and add the functional tests + integration in this PR? Or should I create an issue and do it separately?

**Comment by kennylavender** [comment]
@jwicks31 tagging QE as it would be good to get their input on what they want you to do for the tests.

File: src/Health/HealthController.ts [hunk 3]
```diff
@@ -98,6 +109,12 @@ export class HealthController {
     return pickWalletInfoKeys(await this.bitcoinCore.getWalletInfo())
   }
 
+  private async addBalanceLowBool(walletInfo: WalletInfo): Promise<WalletInfo> {
```
Comment (kennylavender) [inline src/Health/HealthController.ts]: Looks like you got part of the logic unit tested, I think we have more testable logic though. I think we can test it all including the destructuring and adding a new property to a new object if we make a function that takes a wallet and a lowWalletAmount.

```ts
type addWalletIsBalanceLow = (lowBalanceAmount: number) => (wallet: WalletInfo) => WalletInfo
```

We can call that function from our Promise function that knows about the configuration.

Comment (jwicks31) [reply src/Health/HealthController.ts]: Ok cool. I updated this.

File: src/Health/HealthController.test.ts [hunk 4]
```diff
@@ -17,3 +17,41 @@ describe('isStatus200()', async (assert: any) => {
...
+      actual: addWalletIsBalanceBelow2(walletInfo),
+      expected: { ...walletInfo, isBalanceLow: false },
+    })
+  }
+
+  {
+    const walletInfo = walletGen(1)
+    assert({
+      given: 'a lowBalanceAmount higher than balance property of walletInfo',
+      should: 'return walletInfo with correct isBalanceLow bool',
```
Comment (kennylavender) [inline src/Health/HealthController.test.ts]: make should a constant so it's not retyped in all 3 assertions.

**Pull request merged**